src/Adyen/Service/AbstractResource.php

```diff
@@ -261,4 +261,29 @@ public function requestHttp($url, $method = 'get', array $params = null)
         $curlClient = $this->service->getClient()->getHttpClient();
         return $curlClient->requestHttp($this->service, $url, $params, $method);
     }
+
+    /**
+     * @param $url
+     * @param $method
+     * @param array|null $params
+     * @return mixed
+     * @throws AdyenException
+     */
+    public function requestHttpResource($url, $method = 'get', array $params = null)
```

Comment:
You're completely right, my implementation was super clunky. Changed it to the way @AlexandrosMor did it for the management API. My only worry is now that this is kind of breaking change since the simple POST request for paymentlinks is now ```service->paymentLinks->create(...)``` instead of ```service->paymentLinks(...)```. Let me know if you guys think this is okay!

Reply:
I think this is a breaking change indeed, but it seems like this is what we need to do because going forward more and more endpoints on the API will have multiple methods (GET, PATCH, POST).
So how about we refactor all the existing resource models to this approach and release it as a new major version (v14) instead of updating them one by one?

Reply:
Adding new endpoints should not require a breaking change. We can follow the existing convention, introduce 
two methods `retrievePaymentLink` and `updatePaymentLink` to `Adyen\Service\Checkout` service. Add respective resource classes as well and we are done.

Reply:
Fair point. But in that case we would have something of the likes of ```service->paymentLinks(...)``` for POST, and also ```service->paymentLinks->update(...)``` and retrieve.

Reply:
@Morerice